azavea/raster-foundry
# Connect imported scenes to project

## Overview

This PR adds logic to the `import_geotiff_scenes` DAG to, if `projectId` is specified on an import, add any resultant scenes to that project. To accomplish this, a new field, `projectId` is added to the `Upload` model. It is a nullable foreign key that when present will indicate to the import process that the step of added the created scenes to a project needs to be taken. We use the `/projects/{uuid}/scenes` endpoint from the task to do this.

It also adjusts the authorization logic on the `/projects/{uuid}/scenes` endpoint to allow for root users (which will now include the airflow user) to add scenes to any project.

Finally, this updates the UI to conditional pass the created project to the upload creation endpoint.

This should smooth out the process of creating a new project and importing imagery into it.

### Checklist

~- [ ] Styleguide updated, if necessary~


### Demo

Not much to demo really.

## Testing Instructions

 * Run all migrations
 * Start the server and make sure the airflow stack is running (`./scripts/server --with-airflow`)
 * From the UI, create a new project and choose 'Import imagery' instead of 'Public imagery'
 * Select some tifs to upload
 * Once they are uploaded, they should eventually be added to the project you just created

Closes #1400
